packages/solid-query/src/createMutation.ts

```diff
@@ -17,11 +18,12 @@ export function createMutation<
   TContext = unknown,
 >(
   options: CreateMutationOptions<TData, TError, TVariables, TContext>,
+  queryClient?: () => QueryClient,
 ): CreateMutationResult<TData, TError, TVariables, TContext> {
-  const queryClient = useQueryClient({ context: options().context })
+  const client = useQueryClient(queryClient?.())
```

Comment:
@ardeora In createBaseQuery we are wrapping useQueryClient with createMemo.
However in createMutation and createQueries we do not do this.
Should we add createMemo to those hooks as well?

Reply:
This is fine for now 😄 The memoed queryClient isn't really fully reactive yet.  I'll create a separate PR to wire in the reactivity for the other queryClients too